AMP-61625: Marketing Analytics Browser (#323)
Amplitude-Developer-Docs/amplitude-dev-center: docs/data/sdks/marketing-analytics-browser/index.md

|`attribution.disabled`| Optional. `boolean` | Disable the attribution tracking, attribution is enabled by default |
|`attribution.excludeReferrers`|  Optional. `string[]` | Exclude the attribution tracking for the provided referrers string |
|`attribution.initialEmptyValue`| Optional. `string` | Reset the `sessionId` on a new campaign, Default value is `EMPTY` |
|`attribution.resetSessionOnNewCampaign`| Optional. `boolean` | Reset the `sessionId` on a new campaign, won't create a new session for new campaign by default |
|`pageViewTracking.trackOn`| Optional. `attribution` or `() => boolean` | `attribution` - Fire a page view event attribution information changes. `undefined` - Fire a page view event on page load or on history changes for single page application, default behavior. `() => boolean` - Fire a page view events based on a `trackOn` functions|
|`attribution.pageViewTracking.trackHistoryChanges`  | Optional. `pathOnly` or `all` | Track the page view only on the path changes, track `all` URL changes by default|

--8<-- "includes/sdk-ts-browser/basic-ts-functions.md"

> --8<-- "includes/sdk-ts-browser/marketing-analystic.md"

Reviewer: is "analystic" the right term? If this is a typo, we should fix the filename before we ship.

---
title: Marketing Analytics Browser
description: The Amplitude Marketing Analytics Browser SDK Installation & Quick Start guide.
icon: simple/typescript
---

![npm version](https://badge.fury.io/js/@amplitude%2Fmarketing-analytics-browser.svg)

The Marketing Analytics Browser SDK extends the Browser SDK to identify users and events based on marketing channels. This library is open-source, check it out on [GitHub](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/marketing-analytics-browser).

!!!info "Marketing Analytics Browser SDK Resources"
    [:material-github: Github](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/marketing-analytics-browser) · [:material-code-tags-check: Releases](https://github.com/amplitude/Amplitude-TypeScript/releases?q=marketing-analytics-browser&expanded=true) · [:material-book: API Reference](https://amplitude.github.io/Amplitude-TypeScript/modules/_amplitude_marketing_analytics_browser.html)

!!!note "Marketing Analytics Browser versus the Typescript Broswer SDK"

>     The Marketing Analytics Browser SDK extends the Typescript Browser SDK with automatically web attribution and page view tracking. This doc only includes the configuration related with web attribution and page view tracking. For other functionalities please check the [Typescript Browser](../typescript-browser).

Reviewer 2: ```suggestion
    The Marketing Analytics Browser SDK extends the Typescript Browser SDK with automatic web attribution and page view tracking. This doc only includes the configuration related with web attribution and page view tracking. For other functionalities please check the [Typescript Browser](../typescript-browser).
```

amplitude.init("YOUR_API_KEY_HERE");
</script>
```

## Usage

The marketing analytics browser has the same functionalities as the typescript browser SDK. For the basic usage, please check [here](../typescript-browser/).

### Configuration

> Beside the [basic configuration options](../typescript-browser/#configuration), the Marketing Analytics Browser SDK also provide options to configure web attribution and page view tracking.

Reviewer: ```suggestion
In addition to the [basic configuration options](../typescript-browser/#configuration), the Marketing Analytics Browser SDK also has options to configure web attribution and page view tracking.
```

---
title: Marketing Analytics Browser
description: The Amplitude Marketing Analytics Browser SDK Installation & Quick Start guide.
icon: simple/typescript
---

![npm version](https://badge.fury.io/js/@amplitude%2Fmarketing-analytics-browser.svg)

The Marketing Analytics Browser SDK extends the Browser SDK to identify users/events based on marketing channels. This library is open-source, check it out on [GitHub](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/marketing-analytics-browser).

!!!info "Marketing Analytics Browser SDK Resources"
    [:material-github: Github](https://github.com/amplitude/Amplitude-TypeScript/tree/main/packages/marketing-analytics-browser) · [:material-code-tags-check: Releases](https://github.com/amplitude/Amplitude-TypeScript/releases?q=marketing-analytics-browser&expanded=true) · [:material-book: API Reference](https://amplitude.github.io/Amplitude-TypeScript/modules/_amplitude_marketing_analytics_browser.html)

!!!note "Marketing Analytics Browser versus the Typescript Broswer SDK"

>     Marketing Analytics Browser SDK extend the Typescript Broswer SDK but with tracking web attribution and tracking page view support.

Reviewer 2: I agree with @caseyamp I think this section could be updated to include:
1. When should customers use this vs the standard SDK. For fast automatic tracking?
2. A note about this SDK automatically including marketing plugins
3. Link to the standard SDK
4. Link to Plugins

>     Marketing Analytics Browser SDK extend the Typescript Broswer SDK but with tracking web attribution and tracking page view support.

Author: Not talking about plugins here, because we thought if people knew those are plugins, they might install the typescript browser and the two plugins separately. Then we are not able to track the marketing analytics traffic. Because people only install the marketing analytics package will have the `amplitude-ma-ts` context, which we are using for the matrix. Let me know what you think. @justin-fiedler 
cc:  @kevinpagtakhan

>     Marketing Analytics Browser SDK extend the Typescript Broswer SDK but with tracking web attribution and tracking page view support.

Reviewer 3: My impression was the ma sdk was for low-tech customers that want an all in one tracking solution similar to google analytics. If the customer wants to deal with plugins, it's the better long term solution. Also the MA SDK isn't compatible with the ampli cli

>     Marketing Analytics Browser SDK extend the Typescript Broswer SDK but with tracking web attribution and tracking page view support.

Reviewer 2: > Not talking about plugins here, because we thought if people knew those are plugins, they might install the typescript browser and the two plugins separately. Then we are not able to track the marketing analytics traffic. Because people only install the marketing analytics package will have the amplitude-ma-ts context, which we are using for the matrix. Let me know what you think.

Got it. Agree it makes sense to not mention plugins here as not to confuse non-technical users. I provided a suggested update to the note in TS SDK.

amplitude.init("YOUR_API_KEY_HERE");
</script>
```

## Usage

### Initialize the SDK

--8<-- "includes/sdk-httpv2-notice.md"

--8<-- "includes/sdk-ts-browser/init.md"

> ### Configuration

Reviewer 4: The configuration session seems format not rendering well in preview

|`attribution.disabled`| Optional. `boolean` | Disable the attribution tracking, attribution is enabled by default |
|`attribution.excludeReferrers`|  Optional. `string[]` | Exclude the attribution traccking for the provided referrers string |
|`attribution.initialEmptyValue`| Optional. `string` | Reset the sessionId on a new campaign, Default value is `EMPTY` |
|`attribution.resetSessionOnNewCampaign`| Optional. `boolean` | Reset the sessionId on a new campaign, will not create a new session for new campaign by default |
|`pageViewTracking.trackOn`| Optional. `attribution` or `() => boolean` | `attribution` - Fire a page view event attribution information changes. `undefined` - Fire a page view event on page load or on history changes for single page application, default behavior. `() => boolean` - Fire a page view events based on a trackOn functions|
|`attribution.pageViewTracking.trackHistoryChanges`  | Optional. `pathOnly` or `all` | Track the page view only on the path changes, track `all` url changes by default|

> --8<-- "includes/sdk-ts-browser/basic-ts-functions.md"

Reviewer 4: When looking at the page, it kind feels long one and the Web Attribution part falls on the bottom, I wonder if we could leave the basic-ts-functions out and only the ones with marketing analytics, so we don't duplicate the docs. Any thoughts? cc @kevinpagtakhan @dnshi

Author: Discuss with @kevinpagtakhan about this. His point is customers will try to seek all the information on one page. But I do think market analytics stuff is not stand out. Maybe we can move to the first section?
